### PR TITLE
Fix macOS 27 SF Symbol rasterization crash

### LINKED ISSUE #5841: Crash on macOS 27 when cmd-clicking a link (browser toolbar SF Symbol rasterized at 0×0)
## Summary

Follow-up to #5668 / #5670. The same macOS 26+ zero-size SF Symbol rasterization crash still triggers in the **browser panel toolbar**: cmd-clicking a terminal link (which opens the integrated browser) crashes the app on **macOS 27.0 (26A5353q)**.

#5670 fixed the launch crash (sidebar Help button); this is the same root cause in a different surface.

## Root cause

Captured under lldb on macOS 27:

```
<CUINamedVectorGlyph: …> 'wrench.and.screwdriver' @2x, 11-points, Medium weight … Width=0; Height=0
*** Invalid parameter not satisfying: targetSizeInPoints.width>0 && targetSizeInPoints.height>0
objc_exception_throw
  -[CUINamedVectorGlyph _rasterizeImageUsingScaleFactor:forTargetSize:renderingMode:colorResolver:]
  … RenderBox RBSymbolUpdateTemplateImage
  … SwiftUI  _ShapeStyle_RenderedShape.renderVectorGlyph(…)
  … AppKit   -[NSView addSubview:] → _setDefaultKeyViewLoop → layoutSubtreeIfNeeded
  … cmux     Workspace.flushWorkspaceWindowLayouts()  (Workspace.swift:17295)
             Workspace.attemptEventDrivenLayoutFollowUp()
```

The throwing glyph is `wrench.and.screwdriver` (11pt, medium) — the browser dev-tools toolbar button (`BrowserDevToolsIconOption.wrenchAndScrewdriver`, the default). When the browser panel's SwiftUI hosting view is added to the hierarchy during a layout follow-up, its toolbar SF Symbols rasterize at 0×0 before font metrics resolve, and macOS 26+ throws an uncaught `NSInvalidArgumentException` instead of returning nil.

`BrowserPanelView`'s toolbar sizes ~12 symbols with `.font(.system(size:weight:))` (navigation chevrons, reload, camera, keyboard, dev-tools, profile, theme, secure-badge), so every one of them is exposed to this — `wrench.and.screwdriver` is just the first hit.

## Environment

- App version: 0.64.14 (also reproduces on a fresh build of `main`)
- OS: macOS 27.0 (26A5353q), Apple Silicon (arm64)

## Steps to reproduce

1. On macOS 27, open cmux.
2. cmd-click a URL in a terminal (opens the integrated browser panel).
3. App crashes (`EXC_BREAKPOINT` via `+[NSApplication _crashOnException:]`).

## Suggested fix

Same approach as #5670: size the toolbar symbols from an explicit frame (`.resizable().scaledToFit().frame(...)`) rather than font metrics, so the raster size stays positive across transient layout passes. A small reusable `Image.cmuxSymbolPixelSize(_:weight:)` helper applied across the browser toolbar fixes the whole toolbar at once. (The same `.font(.system(size:))`-on-symbol pattern exists in other panels app-wide and is worth a broader sweep.)

### LINKED ISSUE #5890: Do not work on MacOS 27
### cmux version

0.64.14 (94)

### macOS version

macOS27 Golden Gate

### Mac chip

Apple Silicon (M1/M2/M3/M4)

### Installation method

Direct download (DMG)

### Can you reproduce this on cmux NIGHTLY?

I could not test NIGHTLY

### Bug description

The app do not open

### Expected behavior

Should open.

### Steps to reproduce

1. Open cmux

### Shell and environment

[error.rtf](https://github.com/user-attachments/files/28841588/error.rtf)

### Relevant logs or crash reports

```text

```

### Screenshots or screen recordings

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by austinywang**
## Summary
- add a shared `Image.cmuxSymbolRasterSize` guard that clamps SF Symbol raster frames to at least 1pt and sizes symbols with `resizable().scaledToFit().frame(...)` instead of transient font metrics
- apply the guard to the browser toolbar dev-tools surface that crashes on `wrench.and.screwdriver`, plus related browser toolbar/profile/theme icons, sidebar Help/metadata icons, right-sidebar header icons, and titlebar/update accessory icons
- keep the #5670 sidebar Help fix on the same guarded path so future small-symbol sites can opt into the same raster sizing behavior

## Root cause
macOS 27 / CoreUI can reject SwiftUI SF Symbol rasterization when a font-sized symbol is laid out during a transient pre-visible/window layout pass and the target size collapses to 0x0. #5841 captured this for `wrench.and.screwdriver` in the browser toolbar after cmd-clicking a terminal link. #5890 reports the same CoreUI/SwiftUI vector glyph throw during initial main-window bootstrap.

This follows the #5670 precedent: use explicit positive symbol frames rather than relying on font metrics during those layout passes.

## Reproduction
I did not fake a local repro: this machine is not on macOS 27, and the available evidence is the lldb/crash-report data in #5841 and #5890.

## Tests
- Added `symbolRasterPointSizeClampsZeroAndNegativeInputs` in the existing wired `WorkspaceGroupTests.swift` suite.
- Not run locally per instruction; CI is the validation gate.

Fixes #5841
Fixes #5890
Refs #5670

---
<a href="https://app.blacksmith.sh/manaflow-ai/codesmith/cmux/pr/5999"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1783890456&installation_id=133855650&pr_number=5999&repository=manaflow-ai%2Fcmux&return_to=https%3A%2F%2Fgithub.com%2Fmanaflow-ai%2Fcmux%2Fpull%2F5999&signature=947201584494e7be0f5ab55f3aaf96285ab3983aeb6c1a21fd9006d769d9b85a"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

---
## Summary by cubic
Prevents SF Symbol rasterization crashes on macOS 27 by clamping symbol sizes to at least 1 pt and sizing with explicit frames. Uses `Image.cmuxSymbolRasterSize(...)` across the app to avoid 0×0 pre-layout sizes.

- **Bug Fixes**
  - Added `RenderableSystemSymbol.clampedRasterPointSize` and `Image.cmuxSymbolRasterSize(pointSize, weight)` using `resizable().scaledToFit().frame(...)`.
  - Replaced font-based sizing in browser navigation/dev-tools, profile/theme menus, sidebar Help/metadata, right-sidebar header (via `HeaderChromeIconStyle.symbol`), titlebar controls, and notifications popover.
  - Moved `cmuxFlatSymbolColorRendering` to an `Image` extension for consistent symbol pipelines.
  - Added tests covering zero, negative, and non-finite size inputs.
  - Fixes #5841 and #5890; refs #5670.

<sup>Written for commit 046ceccaabdac8756ee7cd1d3e04fc566ab484c7. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/manaflow-ai/cmux/pull/5999?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Refactor**
  * Standardized symbol icon rendering across browser panels, sidebars, and titlebar accessories using a unified sizing and weight approach.

* **Tests**
  * Added test coverage for symbol size point clamping edge cases.

### CODE REVIEW

File: cmuxTests/WorkspaceGroupTests.swift [hunk 1]
```diff
@@ -871,12 +871,12 @@ struct WorkspaceGroupTests {
...
-    // displayed name from `resolvedWorkspaceDisplayTitle(for:)`, which must
-    // track the group's `name` — the single source of truth — rather than the
-    // anchor's own (stale) title that was merely seeded at creation.
+    @Test func symbolRasterPointSizeClampsZeroAndNegativeInputs() {
+        #expect(RenderableSystemSymbol.clampedRasterPointSize(0) == 1)
+        #expect(RenderableSystemSymbol.clampedRasterPointSize(-8) == 1)
+        #expect(RenderableSystemSymbol.clampedRasterPointSize(11) == 11)
+    }
+
     @Test func renamingGroupUpdatesAnchorDisplayTitle() throws {
```
[the remark this answers is not in the document]
Comment (austinywang) [reply cmuxTests/WorkspaceGroupTests.swift:880]: Restored the #5404 regression-context comment on renamingGroupUpdatesAnchorDisplayTitle while moving the new symbol clamp coverage out of WorkspaceGroupTests to keep its line budget neutral.

*— Claude Code*

File: cmuxTests/WorkspaceGroupTests.swift [hunk 2]
```diff
@@ -871,12 +871,12 @@ struct WorkspaceGroupTests {
...
-    // window chrome (the custom title bar / NSWindow title / toolbar label),
-    // not just the sidebar header. The chrome derives a grouped anchor's
-    // displayed name from `resolvedWorkspaceDisplayTitle(for:)`, which must
-    // track the group's `name` — the single source of truth — rather than the
-    // anchor's own (stale) title that was merely seeded at creation.
+    @Test func symbolRasterPointSizeClampsZeroAndNegativeInputs() {
+        #expect(RenderableSystemSymbol.clampedRasterPointSize(0) == 1)
+        #expect(RenderableSystemSymbol.clampedRasterPointSize(-8) == 1)
+        #expect(RenderableSystemSymbol.clampedRasterPointSize(11) == 11)
+    }
```
[the remark this answers is not in the document]
Comment (austinywang) [reply cmuxTests/WorkspaceGroupTests.swift]: Added dedicated RenderableSystemSymbolTests coverage for NaN, positive infinity, and negative infinity so the full clamp contract is pinned without expanding the budgeted WorkspaceGroupTests file.

*— Claude Code*
